e2xgrader/apps/api.py

```diff
@@ -140,3 +144,187 @@ def get_task_submissions(self, assignment_id, notebook_id, task_id):
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
+        for i in nb.cells:
+            if i['metadata']['nbgrader']['grade_id'] == cell_id:
+                i['source'] = cell_content
+                fname = "source/" + assignment + "/" + notebook + ".ipynb"
+                with open(fname, 'w') as f:
+                    nbformat.write(nb, f)
+                
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, source = cell_content)
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, checksum = utils.compute_checksum(i))
+
+                return utils.compute_checksum(i)
+
+        return None
+
+    def student_autograde(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        Returns
+        -------
+        None
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
+        grade_id = []
+        autograde_id = []
+        for idx, i in enumerate(nb.cells):
+            if str(nb.cells[idx]['metadata']['nbgrader']['grade']) == 'True':
+                grade_id.append(str(nb.cells[idx]['metadata']['nbgrader']['grade_id']))
+            if str(nb.cells[idx]['metadata']['nbgrader']['grade']) == 'True'\
+            and str(nb.cells[idx]['metadata']['nbgrader']['solution']) == 'False':
+                autograde_id.append(str(nb.cells[idx]['metadata']['nbgrader']['grade_id']))
+
+        src_dir = os.getcwd() + '/gradebook.db'
+        dst_dir = os.getcwd() + '/gradebook_temp.db'
+        shutil.copy(src_dir,dst_dir)
+
+        grading = {}
+        for i in self.students:
+            grading[i.id] = {}
+            for j in grade_id:
+                grade = self.find_grade(grade_cell = j, notebook = 'Assignment_1', assignment = 'Assignment_1', student = i.id)
+                grading[i.id][j] = grade.score
+                self.update_grade(grade_cell = j, notebook = 'Assignment_1', assignment = 'Assignment_1', student = i.id, manual_score = None)
+
+        nbg = NbGraderAPI()
+
+        for i in self.students:
+            nbg.autograde(assignment, i.id)
+
+        for i in grading:
+            grade = self.find_grade(grade_cell = cell_id, notebook = 'Assignment_1', assignment = 'Assignment_1', student = i)
+            grading[i][cell_id] = grade.auto_score
+
+        for i in self.students:
+            for j in grade_id:
+                self.update_grade(grade_cell = j, notebook = 'Assignment_1', assignment = 'Assignment_1', student = i.id, manual_score = float(grading[i.id][j]))
+
+        return None
```

Comment:
We don't return None, we just write ```return``` or nothing

Reply:
Changed the code.